src/util/map/WrappedSetMultiMap.ts

```diff
@@ -0,0 +1,151 @@
+import type { SetMultiMap } from './SetMultiMap';
+
+/**
+ * A {@link SetMultiMap} that uses an input map to store its data.
+ * The map that is provided to the constructor needs to be empty,
+ * and not be interacted with anymore.
+ * Otherwise the `size` parameter of this Map will no longer be correct.
+ *
+ * In case no input Map is provided, the default Map implementation will be used.
+ *
+ * It is required that the value type of this map is not Set or any extension of Set,
+ * otherwise the `set` and `add` functions wil break.
+ */
+export class WrappedSetMultiMap<TKey, TVal> implements SetMultiMap<TKey, TVal> {
```

Comment:
Do we really need a separate interface, or can this just be the only implementation?

Reply:
Modularity!